### PR TITLE
Added side-by-side categorical hist plotting example

### PR DESCRIPTION
**Pull request by 0ctagon**
Added this example, requested multiple times by users.

### CODE REVIEW

File: docs/examples/1d_hist/1d_side_by_side.py
Comment (cyrraz) [inline docs/examples/1d_hist/1d_side_by_side.py:37]: This is bad practice because we loose information in that operation. We should keep working with the boost_histogram objects.

**Review by cyrraz — CHANGES_REQUESTED** [review]
Thanks for the PR, I wrote a couple of comments.

File: docs/examples/1d_hist/1d_side_by_side.py
Comment (0ctagon) [reply docs/examples/1d_hist/1d_side_by_side.py:13]: We can discuss, for now I move it to Other complex examples section. I really like having this plot in the gallery, and it is much easier to do with boost-histogram than matplotlib.

File: docs/examples/1d_hist/1d_side_by_side.py
Comment (0ctagon) [reply docs/examples/1d_hist/1d_side_by_side.py:51]: Yes, it has been tested (and it's very satisfying to test).

File: docs/img/1d_side_by_side.svg
Comment (0ctagon) [reply docs/img/1d_side_by_side.svg:1]: Agreed

File: docs/examples/1d_hist/1d_side_by_side.py
Comment (0ctagon) [reply docs/examples/1d_hist/1d_side_by_side.py:76]: going with this solution for now.

File: docs/examples/1d_hist/1d_side_by_side.py
Comment (0ctagon) [reply docs/examples/1d_hist/1d_side_by_side.py:68]: Entry.
It turns out it is actually not needed, even for low bin values, the text is correctly placed on top of the bin.

File: docs/examples/1d_hist/1d_side_by_side.py
Comment (0ctagon) [reply docs/examples/1d_hist/1d_side_by_side.py:63]: Which unwanted results exactly? I get beautiful plots until N>10000, at this point the user can either increase the bin width, the figsize or write with some `10k` notation.

File: docs/examples/1d_hist/1d_side_by_side.py
Comment (0ctagon) [reply docs/examples/1d_hist/1d_side_by_side.py:69]: For weighted histograms, it displays the weighted bin entries. The user only need to change the `ylabel` to `Weighted entries`, but he doesn't have to modify this `ax.text` section. What do you exactly mean by "This only works for an unweighted histogram"?

**Comment by 0ctagon** [comment]
https://www.1001fonts.com/download/latin-modern-roman.zip seems down for the moment, so the pytests are failing.

File: docs/examples/1d_hist/1d_side_by_side.py
Comment (cyrraz) [reply docs/examples/1d_hist/1d_side_by_side.py]: I mean that we cast the height to int, which is wrong for a weighted histogram

File: docs/examples/advanced/1d_side_by_side.py
Comment (cyrraz) [inline docs/examples/advanced/1d_side_by_side.py:69]: We cannot assume height is a int in general (see answer in the thread), this is one reason why I wanted to remove this part, but this is minor, we could keep it, your call.

**Review by cyrraz — APPROVED** [review]

**Comment by cyrraz** [comment]
@0ctagon Approved, but please check my comment before merging.

**Pull request merged**
